brave/brave-browser
# Detect changes in v8 and create patch files under patches/v8

This stores patch files for v8 under a `patches/v8` directory. Existing chromium patches are left as-is under `patches/`.

An alternative I considered was to use a `patches-v8` and `patches-chromium` directory. However, I decided on this simpler strategy for now.

The corresponding brave-core `apply-patches.py` script does _not_ need to be modified with this strategy.

Refactors lib/updatePatches.js to become a re-usable library function which is now used by both chromium and v8.

Fix #4282

## Test Plan:

### Test existing chromium functionality
- Run `npm run update_patches`, check that output in brave-core's patch/ directory remains unchanged
- Test simple chromium modification
- Run `npm run update_patches`, check that patch file is created or modified appropriately, as before
- Reset a chromium file to chromium state
- Run `npm run update_patches`, check that the appropriate patch file is removed

### Test new v8 functionality
- Directory at `patches/v8` should _not_ exist (probably)
- Test simple v8 modification
- Run `npm run update_patches`, check that patch file is created or modified appropriately, under patches/v8 directory
- Reset a v8 file to original v8 state
- Run `npm run update_patches`, check that the appropriate v8 patch file is removed

### Test application of patches works succesfully
- Test simple v8 modification
- Run `npm run update_patches`, check that patch file is created or modified appropriately, under patches/v8 directory
- Reset a v8 file to original v8 state
- Run `npm run apply_patches`, check that v8 modification comes back

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on



## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
